src/ssr.rs

```diff
@@ -1,156 +1,187 @@
     }
 
-    fn render(source: String, entry_point: &str, params: Option<&str>) -> String {
-        //The isolate rapresente an isolated instance of the v8 engine
-        //Object from one isolate must not be used in other isolates.
-        let isolate = &mut v8::Isolate::new(Default::default());
-
-        //A stack-allocated class that governs a number of local handles.
-        let handle_scope = &mut v8::HandleScope::new(isolate);
+    pub fn render(&mut self, params: &str) -> String {
+        let mut handle_scope = v8::HandleScope::new(&mut self.isolate);
 
-        //A sandboxed execution context with its own set of built-in objects and functions.
-        let context = v8::Context::new(handle_scope);
+        // A sandboxed execution context with its own set of built-in objects and functions.
+        let context = v8::Context::new(&mut handle_scope);
 
-        //Stack-allocated class which sets the execution context for all operations executed within a local scope.
-        let scope = &mut v8::ContextScope::new(handle_scope, context);
+        // Stack-allocated class which sets the execution context for all operations executed within a local scope.
+        let mut scope = v8::ContextScope::new(&mut handle_scope, context);
 
-        let code = v8::String::new(scope, &format!("{};{}", source, entry_point))
-            .expect("Invalid JS: Strings are needed");
+        let code = v8::String::new(
+            &mut scope,
+            // We add the "entry_point" to the end of the file so
+            // we can get the value of the Webpack bundle as output
+            // when the script is run. See: "object" below which
+            // refers to the output of the script.
+            &format!("{};{}", self.source_code, self.entry_point),
+        )
+        .expect("Invalid JS: Strings are needed");
 
-        let script = v8::Script::compile(scope, code, None)
+        let script = v8::Script::compile(&mut scope, code, None)
             .expect("Invalid JS: There aren't runnable scripts");
 
-        let exports = script
-            .run(scope)
+        let script_exports = script
+            .run(&mut scope)
             .expect("Invalid JS: Missing entry point. Is the bundle exported as a variable?");
 
-        let object = exports
-            .to_object(scope)
-            .expect("Invalid JS: There are no objects");
+        let object = script_exports
+            .to_object(&mut scope)
+            .expect("Invalid JS: entry_point not found. Are you sure you used the right value for entry_point?");
 
-        let fn_map = Self::create_fn_map(scope, object);
+        let root_export_v8 = v8::String::new(&mut scope, &self.root_export)
+            .expect("Failed to allocate string.")
+            .into();
 
-        let params: v8::Local<v8::Value> = match v8::String::new(scope, params.unwrap_or("")) {
-            Some(s) => s.into(),
-            None => v8::undefined(scope).into(),
+        let function = unsafe {
```

Comment:
Good work here. Should we think to both solutions? With and without multiple entry points

Reply:
Yeah I like the idea of supporting a second entry point. In the same vein, I wonder if there's a way to support different exports (not just variable exports).

Reply:
In fact it could be interesting find out if it's possible run a global function like 
```javascript
(() => {
// Render logic
})()
```